sonar-scanner-engine/src/main/java/org/sonar/batch/DefaultFileLinesContextFactory.java

```diff
@@ -19,34 +19,50 @@
  */
 package org.sonar.batch;
 
+import com.google.common.base.Preconditions;
+import org.sonar.api.batch.SensorContext;
 import org.sonar.api.batch.SonarIndex;
 import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.measure.MetricFinder;
 import org.sonar.api.measures.FileLinesContext;
 import org.sonar.api.measures.FileLinesContextFactory;
 import org.sonar.api.resources.File;
 import org.sonar.api.resources.Resource;
+import org.sonar.api.resources.ResourceUtils;
+import org.sonar.batch.index.BatchComponentCache;
+import org.sonar.batch.scan.measure.MeasureCache;
 
 public class DefaultFileLinesContextFactory implements FileLinesContextFactory {
 
   private final SonarIndex index;
+  private final SensorContext sensorContext;
+  private final MetricFinder metricFinder;
+  private final MeasureCache measureCache;
+  private final BatchComponentCache scannerComponentCache;
 
-  public DefaultFileLinesContextFactory(SonarIndex index) {
+  public DefaultFileLinesContextFactory(SonarIndex index, SensorContext sensorContext, MetricFinder metricFinder, BatchComponentCache scannerComponentCache,
+    MeasureCache measureCache) {
     this.index = index;
+    this.sensorContext = sensorContext;
+    this.metricFinder = metricFinder;
+    this.scannerComponentCache = scannerComponentCache;
+    this.measureCache = measureCache;
   }
 
   @Override
-  public FileLinesContext createFor(Resource model) {
+  public FileLinesContext createFor(Resource resource) {
+    Preconditions.checkArgument(ResourceUtils.isFile(resource));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "ResourceUtils"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)